flexget/components/managed_lists/lists/ombi_list.py

```diff
@@ -0,0 +1,548 @@
+        data['theMovieDbId'] = tmdb_id
+
+        return OmbiTv(base_url, auth, data)
+
+
+class OmbiSet(MutableSet):
+    """The schema for the Ombi managed list."""
+
+    supported_ids = ['imdb_id', 'tmdb_id', 'ombi_id']
+    schema = {
+        'type': 'object',
+        'properties': {
+            'url': {'type': 'string'},
+            'api_key': {'type': 'string'},
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'type': {'type': 'string', 'enum': ['shows', 'seasons', 'episodes', 'movies']},
+            'only_approved': {'type': 'boolean', 'default': True},
+            'include_available': {'type': 'boolean', 'default': False},
+            'include_year': {'type': 'boolean', 'default': False},
+            'include_ep_title': {'type': 'boolean', 'default': False},
+        },
+        'oneOf': [{'required': ['username', 'password']}, {'required': ['api_key']}],
+        'required': ['url', 'type'],
+        'additionalProperties': False,
+    }
+
+    @property
+    def immutable(self):
+        return False
+
+    def __init__(self, config: Dict[str, Any]):
+        self.config = config
+        self._items = None
+        self._cached_items = None
+
+    def __iter__(self):
+        return iter(self.items)
+
+    def __len__(self):
+        return len(self.items)
+
+    def add(self, entry: Entry):
+
+        log.info(f"Adding {entry['title']} to ombi_list.")
+
+        log.debug(f"Getting OMBI entry for {entry['title']}.")
+
+        ombi_entry = self._get_ombi_entry(entry)
+
+        if not ombi_entry:
+            log.error(f"Failed to find OMBI entry for {entry['title']}.")
+            return
+
+        ombi_entry.mark_requested()
+
+        log.info(f"{entry['title']} was added to ombi_list.")
+
+    def __ior__(self, entries: List[Entry]):
+
+        for entry in entries:
+            self.add(entry)
+
+    def discard(self, entry: Entry):
+        """Removes an entry from OMBI by marking it as available.
+
+        Args:
+            entry (Entry): An item from a task.
+        """
+
+        log.info(f"Removing {entry['title']} from ombi_list.")
+
+        log.debug(f"Getting OMBI entry for {entry['title']}.")
+
+        ombi_entry = self._get_ombi_entry(entry)
+
+        if not ombi_entry:
+            log.error(f"Failed to find OMBI entry for {entry['title']}.")
+            return
+
+        ombi_entry.mark_requested()
+        ombi_entry.mark_available()
+
+        log.info(f"{entry['title']} was removed from ombi_list.")
+
+    def __isub__(self, entries: List[Entry]):
+
+        for entry in entries:
+            self.discard(entry)
+
+    def _find_entry(self, entry: Entry):
+
+        if "tmdb_id" not in entry:
+            log.warning(
+                f"{entry['title']} is missing the tmdb_id, consider using tmdb_lookup plugin."
+            )
+            return None
+
+        for item in self.items:
+            if item['tmdb_id'] == entry['tmdb_id']:
```

Comment:
this crashed here because there was no tmdb_id in item

maybe:
`    def _find_entry(self, entry: Entry):
        any_key_found = False  
        for key in ('tmdb_id', 'tvdb_id', 'imdb_id'):
            if key in entry and entry[key] is not None:
                any_key_found = True               
            else:
              continue             

            for item in self.items:
                if key in item and item[key] == entry[key]:
                    return item                                 

        if not any_key_found:
            log.warning(
                f"{entry['title']} is missing any useful identifier, consider using tmdb_lookup plugin."
            )                              
        return None `